### PR TITLE
Add functionality test

### PR DESCRIPTION
**Pull request by pengz1**
**Background**
Since UCS-service is an independent service, we need independent functional test for it.

**Details**
* Add unit_tests and function_tests in tests, move unittest cases to unit_tests
* Move UCS API tests from FIT to function_tests
* Add readme to run functional tests
* Update description on run unit_tests

### CODE REVIEW

**Comment by pengz1** [comment]
@PengTian0 @mcgG @bbcyyb @anhou

**Comment by RackHDCI** [comment]
The Copyright Check has failed!
The files listed below do not have a valid copyright statement:
- .travis.yml - tests/function_tests/README.md - tests/function_tests/__init__.py - tests/function_tests/requirements.txt - tests/unit_tests/controllers_tests/__init__.py - tests/unit_tests/service_tests/__init__.py - tests/unit_tests/tasks_tests/__init__.py - tests/unit_tests/util_tests/__init__.py

File: tests/function_tests/run_test.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+if [ -n $3 ]
+then
+    ucs_pass=$3
+fi
+
+virtualenv .venv
+source .venv/bin/activate
+sudo pip install -r requirements.txt
+export UCSCONFIG="{\"ucs-host\": \"$1\", \"ucs-pass\": \"$ucs_pass\", \"ucs-user\": \"$ucs_user\"}"
+nosetests -v --with-nosedep test_ucs_api
```
Comment (PengTian0) [inline tests/function_tests/run_test.sh]: How about pass the ucsconfig path as a parameter

File: tests/function_tests/test_common.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+)
+sys.path.append("/".join((os.path.split(os.path.realpath(__file__))[0]).split('/')[:-2]))
+
+from util import util  # noqa: E402
+
+UCSM_CONFIG = json.loads(os.getenv(
+    'UCSCONFIG',
+    '''
+    {
+        "ucs-host": "10.240.16.144",
```
Comment (PengTian0) [inline tests/function_tests/test_common.py]: It's not secure to expose the IP

Comment (pengz1) [reply tests/function_tests/test_common.py]: Good finding, I will remove default UCSM configure.

File: tests/function_tests/run_test.sh (hunk 1 above)
Comment (pengz1) [reply tests/function_tests/run_test.sh]: It is not easy to pass parameters and it is not suggested by nose

File: tests/function_tests/test_common.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+'''
+  Copyright 2017, Dell, Inc.
+
+  Author(s):
```
Comment (anhou) [inline tests/function_tests/test_common.py]: it's empty

File: tests/function_tests/test_ucs_api.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
+'''
+Copyright 2017, Dell, Inc.
```
Comment (anhou) [inline tests/function_tests/test_ucs_api.py]: not the same with others

File: tests/function_tests/test_ucs_api.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+Author(s):
```
Comment (anhou) [inline tests/function_tests/test_ucs_api.py]: the same with above

**Pull request merged**